### ISSUE TITLE
FB7367637: SwiftUI `.popover()` in a List cannot be dismissed from inside the popover view on macOS

### ISSUE THREAD

**Issue by sindresorhus** [feedback-assistant/reports#50]
- **Date:** 2019-10-11
- **Resolution:** Fixed
- **Area:** SwiftUI Framework
- **OS:** macOS 10.15
- **Type:** Incorrect/Unexpected Behavior

## Description

I have a `List` of buttons. When a button is clicked, a popover is shown from that list item. Inside the popover, there's a "Dismiss" button that is supposed to dismiss the popover, but that doesn't work.

I have included a test project that reproduces the issue. I'm also including the code inline as it's not much:

```swift
import SwiftUI

// Reproduce: Click one of the "Click me" button and then click "Dismiss" in the popover. Notice how the popover is not dismissed.

struct PopoverView: View {
	@Environment(\.presentationMode) var presentationMode

	var body: some View {
		Button(action: {
			self.presentationMode.wrappedValue.dismiss()
		}) {
			Text("Dismiss")
				.padding()
		}
	}
}

struct ListItemView: View {
	let index: Int
	@State var showPopover = false

	var body: some View {
		Group {
			Button(action: {
				self.showPopover = true
			}) {
				Text("Click me \(index)")
			}
			Text("\(index)")
				.popover(isPresented: self.$showPopover) {
					PopoverView()
				}
		}
	}
}

struct ContentView: View {
    var body: some View {
		VStack {
			List {
				ForEach(0...5, id: \.self) { index in
					ListItemView(index: index)
				}
			}
		}
			.frame(width: 400, height: 400)
    }
}
```

## Files

[swiftui-dismiss-popover-in-list-testcase.zip](https://github.com/feedback-assistant/reports/files/3715930/swiftui-dismiss-popover-in-list-testcase.zip)

**Comment by sindresorhus** [comment]
Confirmed fixed in macOS 10.15.1.

**Comment by abara-kedavra** [comment]
The same if you have "self.presentationMode.wrappedValue.dismiss()" inside button that appears in a modal view later, after some event.
